### PR TITLE
Add ExitCode to the "Exited" message

### LINKED ISSUE #169: Write Nonzero Exit Code After "Exited"?
Could the exit (aka return aka result) code from a process be logged, maybe only if nonzero, after `Exited`?

Not all components (run directly or indirectly by a `Procfile`) log the reason they're exiting. For example, I spent quite awhile tracking down what turned out to be a case of `~/.nvm/nvm.sh` returning exit code `3` with no other explanation — but even finding that took awhile, because the output of Overmind didn't make clear to me _why_ it decided to shut down (`Interrupting...`) all the processes.

Seeing several processes simply logged as `Exited`, followed by `Interrupting...` of a process, can be confusing — especially if the interrupted process throws a semi-random error, as a result of being interrupted, such as a stack trace.

### PR DESCRIPTION
**Pull request by prog-supdex**
Resolve #169 

Add the Exit code to the "Exited"

### CODE REVIEW

File: start/process.go [hunk 1]
```diff
@@ -209,3 +210,14 @@ func (p *process) respawn() {
 	p.waitPid()
 	p.output.WriteBoldLinef(p, "Restarted with pid %v...", p.pid)
 }
+
+func (p *process) reportExitCode() {
+	exitCode := p.tmux.ExitCode()
```
Comment (DarthSim) [inline start/process.go:215]: `tmuxClient.ExitCode()` returns the maximum exit code among all panes. It's not applicable here

File: start/process.go [hunk 2]
```diff
@@ -209,3 +210,14 @@ func (p *process) respawn() {
...
+	message := "Exited"
+
+	if exitCode != 0 {
+		message = fmt.Sprintf("Exited with code %d", exitCode)
+	}
```
Comment (Envek) [inline start/process.go:220]: Does it hurt to always print exit code?

```suggestion
	message := fmt.Sprintf("Exited with code %d", exitCode)
```

File: start/process.go (hunk 1 above)
Comment (prog-supdex) [reply start/process.go]: thanks, done!

File: start/process.go (hunk 2 above)
Comment (prog-supdex) [reply start/process.go]: done!

**Review by DarthSim — APPROVED** [review]

**Comment by jcburley** [comment]
Thanks!!

**Pull request merged**